e3sm_diags/driver/arm_diags_driver.py

```diff
@@ -23,179 +27,162 @@
                 )
-                parameter.var_name = getattr(test, "long_name", var)
-                parameter.var_units = getattr(test, "units", var)
 
                 refs = []
 
                 if "armdiags" in ref_name:
                     if region != "sgpc1":
-                        msg = "Diurnal cycle of {} at Site: {} is not supported yet".format(
-                            region, var
+                        raise RuntimeError(
+                            f"Diurnal cycle of {region} at Site: {var} is not "
+                            "supported yet"
                         )
-                        raise RuntimeError(msg)
                     else:
                         ref_file_name = "sgparmdiagsmondiurnalC1.c1.nc"
-
                         ref_file = os.path.join(ref_path, ref_file_name)
-                        ref_data = cdms2.open(ref_file)
+                        ds_ref = xr.open_dataset(ref_file)
 
                         if var == "PRECT":
-                            ref = (
-                                ref_data("pr") * 3600.0 * 24
-                            )  # Converting mm/second to mm/day"
-                            ref.lat = test.lat
-                            ref.lon = test.lon
-                            (
-                                ref_diurnal,
-                                lst,
-                            ) = utils.diurnal_cycle.composite_diurnal_cycle(
-                                ref, season, fft=False
+                            # Converting mm/second to mm/day"
+                            ds_ref[var] = ds_ref["pr"] * 3600.0 * 24
+                            ds_ref["lat"] = ds_test.lat.values
+                            ds_ref["lon"] = ds_test.lon.values
+
+                            ref_diurnal, lst = composite_diurnal_cycle(  # type: ignore
+                                ds_ref, var, season, fft=False
                             )
-                            if hasattr(ref, "standard_name"):
-                                ref.long_name = ref.standard_name
 
                             ref = ref_diurnal
 
                 else:
-                    ref_data = utils.dataset.Dataset(parameter, ref=True)
-                    ref = ref_data.get_timeseries_variable(var, single_point=True)
-                    ref.lat = test_data.get_static_variable("lat", var)
-                    ref.lon = test_data.get_static_variable("lon", var)
-                    ref_diurnal, lst = utils.diurnal_cycle.composite_diurnal_cycle(
-                        ref, season, fft=False
+                    ref_data = Dataset(parameter, data_type="ref")
+                    ds_ref = ref_data.get_time_series_dataset(var, single_point=True)
+
+                    # FIXME: ref is not set to ref_diurnal here
+                    ref_diurnal, lst = composite_diurnal_cycle(  # type: ignore
+                        ds_ref, var, season, fft=False
                     )
```

Comment:
Is this FIXME: comment still relevant? 

I noticed that ref is never updated to ref_diurnal here. Should this be fixed?

Reply:
`ref = ref_diurnal` should be added here